crates/server/src/api/mod.rs

```diff
@@ -19,11 +24,19 @@ pub fn create_router(
     temp_dir: PathBuf,
     files_dir: PathBuf,
     content_policy: Option<Arc<dyn ContentPolicy>>,
+    authorized_keys: Option<HashMap<String, HashSet<KeyID>>>,
```

Comment:
At some point I would like policies to be a `Vec<dyn Policy>`-style construction where this namespace control policy isn't special.

In this flow, the `Policy` trait would either
* have methods for each of the possible things it could have an opinion on,
* would have getters/factories that optionally return sub-policies for specific areas,
* or would actually be an `enum` such that each policy refers to a single area and you just add multiple policies to the list if you need to refer to multiple areas.

Reply:
I moved this out of `Config` and into an `AuthorizedKeyPolicy` implementing the logic.

The mechanism is minimal right now, but we can expand on the design in future work.